### PR TITLE
Diving patch

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Reduces diving range, increased standup length, adds injury to diving into windows.

## Why It's Good For The Game

No more prisoners breaking out with their heads (uninjured), sliding is no longer superior to running (exception: parkour perk).

## Changelog
:cl:
balance: diving range decreased
balance: Titanfall removed
balance: diving into windows and doors takes longer to break them, damages the jumper
/:cl:

### CODE REVIEW

**Comment by Iglpigl** [comment]
Atomize your PRs.

**Comment by Humonitarian** [comment]
> Atomize your PRs.

https://media.discordapp.net/attachments/892135257334251630/933320438820528158/unknown.png

**Comment by Kegdo** [comment]
i'm fine with everything except the standup length and dive range. why the FUCK should it be so low AND slow?

**Comment by Kegdo** [comment]
(even the length is fine honestly, so it's not faster to dive than to run, but **DONT FUCKING NERF THE REACH JESUS CHRIST**

**Comment by Kegdo** [comment]
image
yes i am fuming and seething

**Comment by Kegdo** [comment]
image
Firstly there is absolutely no reason for this to be made, secondly it will screw up loot and make all these items more common, thirdly, one and possibly more items price changes are direct violation to contributing guidelines, you should try reading them, and lastly, frick you, don't nerf my code without arguing with me first
https://wiki.cev-eris.com/Content_GuidelinesEn

**Comment by Holvr0** [comment]
Ok, let's not do that...

Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.

But don't reduce the dive range - it just makes the diving feature dead on conception.

**Comment by Humonitarian** [comment]
> Ok, let's not do that...
> 
> Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.
> 
> But don't reduce the dive range - it just makes the diving feature dead on conception.

Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?

**Comment by Kegdo** [comment]
> Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?

Unwieldy? Bruv it's literally perfect

**Comment by Iglpigl** [comment]
> > Ok, let's not do that...
> > Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.
> > But don't reduce the dive range - it just makes the diving feature dead on conception.
> 
> Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?

Totally agree with Kegdo on last point, it's perfectly usable in it's current state, I would definitely not call it unwieldy.

**Comment by Kegdo** [comment]
the only time someone agrees with my (correct) opinion is when it really is required, amazing

**Comment by Humonitarian** [comment]
> > > Ok, let's not do that...
> > > Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.
> > > But don't reduce the dive range - it just makes the diving feature dead on conception.
> > 
> > 
> > Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?
> 
> Totally agree with Kegdo on last point, it's perfectly usable in it's current state, I would definitely not call it unwieldy.

Not always unwieldy, but there are situations where the dive is specifically too long to be convenient.

**Comment by Iglpigl** [comment]
> > > > Ok, let's not do that...
> > > > Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.
> > > > But don't reduce the dive range - it just makes the diving feature dead on conception.
> > > 
> > > 
> > > Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?
> > 
> > 
> > Totally agree with Kegdo on last point, it's perfectly usable in it's current state, I would definitely not call it unwieldy.
> 
> Not always unwieldy, but there are situations where the dive is specifically too long to be convenient.

That can be said for anything in the game my g.

**Comment by SirRichardFrancis** [comment]
> > > > > Ok, let's not do that...
> > > > > Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.
> > > > > But don't reduce the dive range - it just makes the diving feature dead on conception.
> > > > 
> > > > 
> > > > Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?
> > > 
> > > 
> > > Totally agree with Kegdo on last point, it's perfectly usable in it's current state, I would definitely not call it unwieldy.
> > 
> > 
> > Not always unwieldy, but there are situations where the dive is specifically too long to be convenient.
> 
> That can be said for anything in the game my g.

Would you kindly stop quoting the whole thing? That's turning one line of text into a wall of text for no reason.

**Comment by SirRichardFrancis** [comment]
> > > > > > Ok, let's not do that...
> > > > > > Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.
> > > > > > But don't reduce the dive range - it just makes the diving feature dead on conception.
> > > > > 
> > > > > 
> > > > > Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?
> > > > 
> > > > 
> > > > Totally agree with Kegdo on last point, it's perfectly usable in it's current state, I would definitely not call it unwieldy.
> > > 
> > > 
> > > Not always unwieldy, but there are situations where the dive is specifically too long to be convenient.
> > 
> > 
> > That can be said for anything in the game my g.
> 
> Would you kindly stop quoting the whole thing? That's turning one line of text into a wall of text for no reason.

Also makes those messages unreadable in discord.

**Comment by Iglpigl** [comment]
You are doing the same thing, but okay.

**Comment by SirRichardFrancis** [comment]
Not usually, but this time yes, doing that to illustrate my point.

**Comment by Holvr0** [comment]
> > > > > Ok, let's not do that...
> > > > > Damaging windows - sure, it's retarded that prisoners can break out by headbashing reinforced windows, so THAT should be out.
> > > > > But don't reduce the dive range - it just makes the diving feature dead on conception.
> > > > 
> > > > 
> > > > Have you tried it? The long dive is unwieldy. What point is there in making the dive longer?
> > > 
> > > 
> > > Totally agree with Kegdo on last point, it's perfectly usable in it's current state, I would definitely not call it unwieldy.
> > 
> > 
> > Not always unwieldy, but there are situations where the dive is specifically too long to be convenient.
> 
> That can be said for anything in the game my g.

Give it variable length depending on whether you're walking or running.
Running - long dive, like we want it.
Walking - short dive, like you want it.

Solved :)

**Comment by Kegdo** [comment]
^ yeah this, we have a macro command for switching move intent too, so that should be a good deal for both sides.

**Comment by Humonitarian** [comment]
Still doesn't fix the issue of Titanfall

**Comment by Kegdo** [comment]
lemme run tests on local to figure out how to properly increase the rest stand up length without making it __one second__ slow

**Comment by Holvr0** [comment]
> Still doesn't fix the issue of Titanfall

Make it so diving eats away some nutrition. Using it for its actual purpose will be ok, but turbo spamming it instead of running will devour your nutrition.

Or just make it so it has a "cooldown" and you can use it like 3 times for free (each "use" regenerates in like 10 seconds), and spamming it beyond that gives pain.

There are many solutions more creative than "nerf it to the ground lol".

**Comment by Kegdo** [comment]
Cooldown is bad game design, but nutrition... That'd probably even make the gamers go to bar. A brilliant idea

File: code/game/objects/structures/window.dm [hunk 1]
```diff
@@ -278,6 +283,20 @@
...
+/obj/structure/window/proc/hit_by_living(var/mob/living/M)
+	var/body_part = pick(BP_HEAD, BP_CHEST, BP_GROIN)
+	visible_message(SPAN_DANGER("[M] slams against \the [src]!"))
+	if(prob(30))
+		M.Weaken(1)
+	M.damage_through_armor(8, BRUTE, body_part, ARMOR_MELEE)
+
+	var/tforce = 15
+	if(reinf) tforce *= 0.25
+	if(health - tforce <= 7 && !reinf)
```
Comment (Kegdo) [inline code/game/objects/structures/window.dm:295]: is this enough to break a basic window in one jump?

Comment (Humonitarian) [reply code/game/objects/structures/window.dm:295]: No. It does the same amount of damage an aggressive grab banging does.

Comment (Holvr0) [reply code/game/objects/structures/window.dm:295]: > Cooldown is bad game design, but nutrition... That'd probably even make the gamers go to bar. A brilliant idea

One thing I'm not sure about is FBPs. Aren't they immune to nutrition?
(Probably a separate PR would be needed, to make FBPs require electricity for nutrition - like, draining batteries or feasting at APCs to regenerate it).

**Comment by Sparrowbird611** [comment]
I got no comment on dive ranges but I'd prefer to see damage removed rather than damaging the user, or at least make it excluded reinforced windows and up.

**Comment by Sparrowbird611** [comment]
> Not always unwieldy, but there are situations where the dive is specifically too long to be convenient.

I would also like to point out that the range is controllable, so I would classify this as a skill issue.

**Comment by Humonitarian** [comment]
> > Not always unwieldy, but there are situations where the dive is specifically too long to be convenient.
> 
> I would also like to point out that the range is controllable, so I would classify this as a skill issue.

Dive range isn't.

**Comment by ValoTheValo** [comment]
keep the normal range. everything else is good

**Comment by Kegdo** [comment]
then, again, make dive range controllable like holvr suggested; 1 tile on walk, 2 tiles on run

**Comment by Kegdo** [comment]
yeah so basically
i would prefer to make these changes myself, while making a compromise for everyone, instead of just nerfing a feature to shit and not taking ANY critique because it's 'unwieldy'; as such, observe as I make the same pr, but by me and better(but only after i buff perks tho)

**Comment by tristain01** [comment]
Firstly as one of the few oldfags remaining I feel I need to speak on this matter, nerfing the range out right kills the ability to jump between z-level gaps, you would be neutering a fun feature, I agree that head butting windows should deal damage , it also shouldn't be instant on reinforced ones BUT do not nerf the range.

**Comment by ValoTheValo** [comment]
>  yeah so basically i would prefer to make these changes myself, while making a compromise for everyone, instead of just nerfing a feature to shit and not taking ANY critique because it's 'unwieldy'; as such, observe as I make the same pr, but by me and better(but only after i buff perks tho)

yeah dont merge it if its gonna be unchanged

**Comment by Humonitarian** [comment]
@Kegdo feel free to use my changes for another PR, code the walk/run thing too since while fairly difficult to swap that quickly, it still allows some solid plays and use of cover.

Also, on hindsight, increasing slide delay from 2 to 2.5 might be better that increasing the standup timer anywhere beyond .8

**Comment by Sparrowbird611** [comment]
Ye increased slide delay is better than standing timer.

**Comment by SirRichardFrancis** [comment]
Cringe.

**Comment by Kegdo** [comment]
cloze the pr pls then it makes me panic every time i see it

**Comment by TheShown911** [comment]
too long didn't read.

**Comment by Humonitarian** [comment]
Kegdo got this covered

**Comment by Humonitarian** [comment]
Kegdo failed

**Comment by Kegdo** [comment]
close it again i am gonna get this covered anyways

**Comment by Kegdo** [comment]
DNM we got the #6985 and because I made that one its automatically better

**Pull request closed without merging**